ndb/tests/unit/test_eventloop.py

```diff
@@ -22,16 +24,304 @@ def test___all__():
     tests.unit.utils.verify___all__(eventloop)
 
 
-def test_add_idle():
-    with pytest.raises(NotImplementedError):
-        eventloop.add_idle()
+def _Event(when=0, what='foo', args=(), kw={}):
+    return eventloop._Event(when, what, args, kw)
 
 
 class TestEventLoop:
     @staticmethod
-    def test_constructor():
+    def _make_one(**attrs):
+        """Use DummyClock"""
+        ev = eventloop.EventLoop(DummyClock())
+        for name, value in attrs.items():
+            setattr(ev, name, value)
+        return ev
+
+    @staticmethod
+    @unittest.mock.patch('google.cloud.ndb.eventloop.time')
+    def test_constructor(time):
+        time.time.return_value = 42
+        ev = eventloop.EventLoop()
+        assert ev.clock.now() == 42
+        ev.clock.sleep(27)
+        time.sleep.assert_called_once_with(27)
+
+        assert ev.current == collections.deque()
+        assert ev.idlers == collections.deque()
+        assert ev.inactive == 0
+        assert ev.queue == []
+        assert ev.rpcs == {}
+
+    def test_clear_all(self):
+        ev = self._make_one()
+        myclock = ev.clock
+        ev.current.append('foo')
+        ev.idlers.append('bar')
+        ev.queue.append('baz')
+        ev.rpcs['qux'] = 'quux'
+        ev.clear()
+        assert not len(ev.current)
+        assert not len(ev.idlers)
+        assert not len(ev.queue)
+        assert not len(ev.rpcs)
+        assert ev.clock is myclock
+
+        # idemptotence (branch coverage)
+        ev.clear()
+        assert not len(ev.current)
+        assert not len(ev.idlers)
+        assert not len(ev.queue)
+        assert not len(ev.rpcs)
+        assert ev.clock is myclock
+
+    def test_clear_current(self):
+        """ For branch coverage. """
```

Comment:
Not sure the docstring is necessary? Also, did you intend to begin and end with a space, i.e. `""" For` vs. `"""For`

Reply:
95a94f4bd0753a77890a3b07648b13845c296ac5